content/posts/windows-containers.md

```diff
@@ -0,0 +1,231 @@
+
+## Infrastructure Setup and Windows Nodepools
+
+Jenkins has the ability to use Kubernetes pods as agents to build and deploy applications thanks to the [Kubernetes plugin](https://github.com/jenkinsci/kubernetes-plugin). In this blog post, we will create a simple declarative pipeline that has Linux and Windows containers as agents in AKS. 
+
+
+
+First, we need to follow [Azure's documentation](https://docs.microsoft.com/en-us/azure/aks/windows-container-cli#before-you-begin) to create the needed infrastructure to be able to deploy Linux and Windows based containers. Please make sure that you review AKS documentation and are aware of the limitations before running this in a production cluster. As explained [here](https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools), the registration for the preview features cannot be unregistered at this moment. 
+
+The [documentation](https://docs.microsoft.com/en-us/azure/aks/windows-container-cli) at a high level goes through the following steps (using the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)):
+
+1. Install aks-preview CLI extension for Azure CLI
+2. Register the Windows preview feature needed for the Windows based containers
+    * As mentioned in the document, the [Multiple Node Pool feature](https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools) is also needed to create a separate Windows node pool.
+3. Create a new resource group (if needed)
+4. Create an AKS cluster
+    * You can use ```--nodepool-name``` with ```aks create cluster``` to name your control plane node pool i.e ```default```
+5. Add a Windows Server node pool 
+    * This will be a node pool for kubernetes pod Windows agents, we can name it ```--name winage```
+
+[Node pools](https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools) give us the possibility to extend our Kubernetes cluster with more types of machines depending on our use and budget (See available [options and default values](https://docs.microsoft.com/en-us/cli/azure/ext/aks-preview/aks/nodepool?view=azure-cli-latest#ext-aks-preview-az-aks-nodepool-add)) for AKS. As an example, we are going to add two more identical pools (one for masters and one for Linux agents) but you can pick different machine sizes and node counts depending on your need (just make sure that the VMs used for Jenkins masters support Premium Storage as Jenkins requires high IOPS for better performance):
+    
+* Linux Jenkins master pool example:
+
+    ```    
+    az aks nodepool add \ 
+    --resource-group myResourceGroup \
+    --cluster-name eastUSAKS \
+    --os-type Linux \
+    --name masters \
+    --node-count 1 \
+    --kubernetes-version 1.14.6 \
+    --node-vm-size Standard_DS2_v2
+    ```
+
+* Linux Jenkins agent pool example:
+
+    ```
+    az aks nodepool add \                                                                 --resource-group myResourceGroup \
```

Comment:
Looks like there's some extra spacing here that's causing `--resource-group` to display out of line

Reply:
Not sure if I fixed this error. I wasn't sure how ``` ---resource-group ``` was out of line. I see it under ```az aks nodepool add \ ``` in the same block.